src/meltano/core/plugin_install_service.py

```diff
@@ -394,6 +394,25 @@ async def install_plugin_async(
             self.status_cb(state)
             return state
 
+    def _requires_install(
+        self,
+        plugin: ProjectPlugin,
+        *,
+        env: t.Mapping[str, str] = None,
+    ) -> bool:
+        try:
+            pip_install_args = get_pip_install_args(
+                self.project,
+                plugin,
+                env,
+                if_missing=EnvVarMissingBehavior.raise_exception,
+            )
+        except EnvironmentVariableNotSetError:
+            return False
```

Comment:
What do you think about logging a warning here?

Reply:
There is already a debug log coming from `expand_env_vars`

https://github.com/meltano/meltano/blob/630eebfed4b45c127f336b9872ebaf031353852b/src/meltano/core/utils/__init__.py#L540-L542

but I suppose we want a more user-centric message like "Environment variable not set for pip_url, skipping install"?

Reply:
> but I suppose we want a more user-centric message like "Environment variable not set for pip_url, skipping install"?

Yeah exactly. They'd get more details by looking at _debug_ logs, but they'd immediately know that they might not get the latest dependencies for that command.

Reply:
145065d8d2917f600a5815d0cd6324846e676024

Reply:
👍